### ISSUE TITLE
GIS integration

### ISSUE THREAD

**Issue by mrceresa** [krABMaga/krABMaga#32]
Dear authors,
thanks for sharing this great toolkit! I'm very interested in using it for city simulations, and for that I would need a sort of GIS integration like loading (and visualizing) shapefiles and fast spatial index (for collisions).

In my past experience with mesa and mesa-gis this would not scale well. 

I was wondering if you could advice on that

Thanks and regards

**Comment by giusdam** [comment]
Dear,
We are pleased to hear of your interest in our toolkit!
Integrating krABMaga with GIS data is a feature we are working on and is the next priority in our plans.
If you want to see some work in Rust regarding GIS, there are some crates like [earthwyrm](https://crates.io/crates/earthwyrm) and [geojson](https://crates.io/crates/geojson).
Regarding the implementation, since krABMaga is based on the Bevy engine that does not support the shape required for the GIS data, we are developing a solution from scratch that may require a bit. But we are optimistic about that.

**Comment by spagnuolocarmine** [comment]
Yes! 
Feel free to write us if we can support you in experimenting with GIS and our tool.

Thank you

**Comment by mrceresa** [comment]
Thanks a lot Giuseppe and Carmine, for your fast reply! Indeed I would be very interested into experimenting with that (it would be a perfect excuse to learn rust :) )
There is a very nice example from mesa that I would like to recreate with krABMaga:
https://github.com/wang-boyu/agents-and-networks-in-python

I tried here to scale up a little using a better rtree implementation:
https://github.com/wang-boyu/agents-and-networks-in-python/pull/2

Let me know if you have any idea/suggestions on how to start reimplementing it with krabm

**Comment by jonititan** [comment]
Hi krABMaga team,

I am also very interested in using krABMaga for a GIS based simulation.
If I send in a pull request with a proposal would you be willing to review it?

I have been experimenting with the [a5 crate](https://a5geo.org/) and loading sim assets from geojson.  I think it will make krABMaga very flexible to go along with its power. 

What do you think?

**Comment by ddevin96** [comment]
hi @jonititan!
Feel free to open a PR! We will look into it!
There is a very early try of implementation in [GIS branch](https://github.com/krABMaga/krABMaga/tree/GIS-bevy-0.13)

**Comment by jonititan** [comment]
I'll have a look thanks.

I have a very shaky A5 implementation working currently.  I'll get back to you shortly with it.  Just need to review again before raising a PR and I'll check it against your branch first.

**Comment by jonititan** [comment]
Currently the bounds are set by using the a5 get disk rather than by setting a bounding box.  Also its slow because my laptop is old.  I haven't profiled it yet.

![Image](https://github.com/user-attachments/assets/21f5aa84-0bbd-4e18-96da-af5b1c72204d)

Once I've refined it a bit and added ability to use bounding box as well I will work on making it so geojson can be used to specify sim asset start positions etc and also to make use of slippy tiles in the visualization.  Probably via [bevy slippy tiles crate](https://crates.io/crates/bevy_slippy_tiles) .

**Comment by jonititan** [comment]
![Image](https://github.com/user-attachments/assets/7cff0b1a-4dee-4367-88ac-2e9dfdd9d89b)

**Comment by ddevin96** [comment]
Looks promising! 
Whenever you feel confident, you can open the PR so we can leave you some feedback
It's still the same bevy version?

**Comment by jonititan** [comment]
Yes still the same bevy version.  Trying to keep the changes minimal.

Just want to add the geojson config and tiles for visualisation

**Comment by ddevin96** [comment]
I have pushed some updates for general management of the repo, as we have started to in our projects
You can also test now with bevy 18.1 if it smooths the implementation 
All new changes are reachable with krabmaga 0.6.*

**Comment by jonititan** [comment]
Thanks, I'll check it out!

**Comment by jonititan** [comment]
<img width="1288" height="654" alt="Image" src="https://github.com/user-attachments/assets/ac253f7e-5caa-4e09-b211-47df2b5b3487" />

Not satisfied with it yet but it loads and saves state as geojsons.
I'm trying to make it so it can just be a trait to be added to agent structs.  Make it easy to add in to peoples simulations.
Storing A5 cell states(grass type in this instance) as point features is a bit silly though.

**Comment by ddevin96** [comment]
I'm not sure if the trait is the correct solution if it is implemented through bevy. 
Maybe it could be easier as a plugin

**Comment by jonititan** [comment]
https://github.com/user-attachments/assets/ce0ffaa1-c189-4cfc-93f9-f45a66841da2

I've made it into a bevy plugin instead as suggested.  I'll do a PR with the GIS grid system changes for krABMaga shortly.
[bevy:a5](https://crates.io/crates/bevy_a5)

**Comment by jonititan** [comment]
I've uploaded an example.  Should I do a PR for it to the examples repo?  Ideally the a5 update would be in cargo by then.
https://github.com/jonititan/cto

**Comment by ddevin96** [comment]
it would be very appreciated @jonititan!

**Comment by jonititan** [comment]
ok.  Will the cargo release get updated to latest?  Or should I keep the example pointing at github for the a5 gis feature?

**Comment by ddevin96** [comment]
When i merged, i published the 0.6.1 release, should be working.

**Comment by jonititan** [comment]
But wasn't this merged after the 0.6.1 release?

**Comment by ddevin96** [comment]
Yes, you were right. now 0.6.2 is available on crates.io
